dvc/repo/plots/template.py

```diff
@@ -106,466 +94,75 @@ def _check_field_exists(data, field):
-                "layer": [
-                    {"mark": "line"},
-                    {
-                        "selection": {
-                            "label": {
-                                "type": "single",
-                                "nearest": True,
-                                "on": "mouseover",
-                                "encodings": ["x"],
-                                "empty": "none",
-                                "clear": "mouseout",
-                            }
-                        },
-                        "mark": "point",
-                        "encoding": {
-                            "opacity": {
-                                "condition": {
-                                    "selection": "label",
-                                    "value": 1,
-                                },
-                                "value": 0,
-                            }
-                        },
-                    },
-                ],
-            },
-            {
-                "transform": [{"filter": {"selection": "label"}}],
-                "layer": [
-                    {
-                        "mark": {"type": "rule", "color": "gray"},
-                        "encoding": {
-                            "x": {
-                                "field": Template.anchor("x"),
-                                "type": "quantitative",
-                            }
-                        },
-                    },
-                    {
-                        "encoding": {
-                            "text": {
-                                "type": "quantitative",
-                                "field": Template.anchor("y"),
-                            },
-                            "x": {
-                                "field": Template.anchor("x"),
-                                "type": "quantitative",
-                            },
-                            "y": {
-                                "field": Template.anchor("y"),
-                                "type": "quantitative",
-                            },
-                        },
-                        "layer": [
-                            {
-                                "mark": {
-                                    "type": "text",
-                                    "align": "left",
-                                    "dx": 5,
-                                    "dy": -5,
-                                },
-                                "encoding": {
-                                    "color": {
-                                        "type": "nominal",
-                                        "field": "rev",
-                                    }
-                                },
-                            }
-                        ],
-                    },
-                ],
-            },
-        ],
-    }
-
-
 class PlotTemplates:
     TEMPLATES_DIR = "plots"
```

Comment:
Why keep this? Plots included in pkg dir + being able to specify custom template path should be enough, and dropping `TEMPLATES_DIR` will ignore outdated templates in `.dvc/plots`.

Reply:
The problem here is what in a case when user modified hers/his plots in `.dvc/plots`? If we drop `.dvc/plots` user will need to specify the path. It's not a big problem but will be annoying. Also `.dvc/plots` seems like a good default place for users to put their own templates if they don't want to clutter their repo with templates dir. On the other hand, repos created pre-this-release will keep using outdated templates which is also not good. 

We can always consider dropping `.dvc/plots` in `3.0`